modules/openapi-generator-bazel/src/defs.bzl

```diff
@@ -0,0 +1,168 @@
+    gen_cmd += " 1>/dev/null"
+    return gen_cmd
+
+def _impl(ctx):
+    jars = _collect_jars(ctx.attr.deps)
+    (cjars, rjars) = (jars.compiletime, jars.runtime)
+
+    declared_dir = ctx.actions.declare_directory("%s" % (ctx.attr.name))
+
+    inputs = [
+        ctx.file.openapi_generator_cli,
+        ctx.file.spec,
+    ] + cjars.to_list() + rjars.to_list()
+
+    # TODO: Convert to run
+    ctx.actions.run_shell(
+        inputs = inputs,
+        command = "mkdir -p {gen_dir}".format(
+            gen_dir = declared_dir.path,
+        ) + " && " + _new_generator_command(ctx, declared_dir, rjars),
+        outputs = [declared_dir],
+        tools = ctx.files._jdk,
+    )
+
+    target = ctx.outputs.codegen.path
+    srcs = declared_dir.path
+    ctx.actions.run(
+        executable = ctx.executable._jar,
+        arguments = ["cMf", target, "-C", srcs, "."],
+        outputs = [ctx.outputs.codegen],
+        inputs = inputs + [declared_dir],
+    )
+
+    return DefaultInfo(files = depset([
+        declared_dir,
+        # TODO: Not all users of this will be interested in a .srcjar output. As such,
+        # it would be nice to break it into two separate providers so that a .srcjar isn't
+        # created unless the caller depends on it.
+        ctx.outputs.codegen,
+    ]))
+
+def _collect_jars(targets):
+    """Compute the runtime and compile-time dependencies from the given targets"""  # noqa
+    compile_jars = depset()
+    runtime_jars = depset()
+    for target in targets:
+        found = False
+        if hasattr(target, "scala"):
+            if hasattr(target.scala.outputs, "ijar"):
+                compile_jars = depset(transitive = [compile_jars, [target.scala.outputs.ijar]])
+            compile_jars = depset(transitive = [compile_jars, target.scala.transitive_compile_exports])
+            runtime_jars = depset(transitive = [runtime_jars, target.scala.transitive_runtime_deps])
+            runtime_jars = depset(transitive = [runtime_jars, target.scala.transitive_runtime_exports])
+            found = True
+        if hasattr(target, "java"):
+            compile_jars = depset(transitive = [compile_jars, target.java.transitive_deps])
+            runtime_jars = depset(transitive = [runtime_jars, target.java.transitive_runtime_deps])
+            found = True
+        if not found:
+            runtime_jars = depset(transitive = [runtime_jars, target.files])
+            compile_jars = depset(transitive = [compile_jars, target.files])
+
+    return struct(compiletime = compile_jars, runtime = runtime_jars)
+
+openapi_generator = rule(
+    attrs = {
+        # downstream dependencies
+        "deps": attr.label_list(),
+        # openapi spec file
+        "spec": attr.label(
+            mandatory = True,
+            allow_single_file = [
+                ".json",
+                ".yaml",
+            ],
+        ),
+        # language to generate
+        "language": attr.string(mandatory = True),
```

Comment:
This option has been deprecated for a while, and removed in more recent versions. l/language has been replaced with g/generator-name in CLI options.

Reply:
Done! I was actually using the -g flag in my invocation, I just didn't reflect that in the bazel rule :p